src/ray/util/signal_test.cc

```diff
@@ -0,0 +1,100 @@
+#include <signal.h>
+#include <cstdlib>
+#include <iostream>
+
+#include "gtest/gtest.h"
+#include "ray/util/logging.h"
+#include "ray/util/signal_handler.h"
+#include "ray/util/util.h"
+
+// This test just print some call stack information.
+namespace ray {
+
+void Sleep() { usleep(100000); }
+
+void TestSendSignal(const std::string &test_name, int signal) {
+  pid_t pid;
+  pid = fork();
+  ASSERT_TRUE(pid >= 0);
+  if (pid == 0) {
+    while (true) {
+      int n = 1000;
+      while (n--)
+        ;
+    }
+  } else {
+    Sleep();
+    RAY_LOG(ERROR) << test_name << ": kill pid " << pid
+                   << " with return value=" << kill(pid, signal);
+    Sleep();
+  }
+}
+
+TEST(SignalTest, SendTermSignal_Unset_Test) {
+  // SignalHandlers will be automatically uninstalled when it is out of scope.
+  DefaultInitShutdown signal_handler_uninstall_wrapper(
+      SignalHandlers::InstallSignalHandler, SignalHandlers::UninstallSignalHandler,
+      "SendTermSignal_Unset_Test", false);
```

Comment:
For readability, how about writing it like this:
```c++
DefaultInitShutdown signal_handler_uninstall_wrapper(
      SignalHandlers::InstallSignalHandler,"SendTermSignal_Unset_Test", false);
signal_handler_uninstall_wrapper.RegisterShundownHandler(SignalHandlers::UninstallSignalHandler);
```
or 
```c++
DefaultInitShutdown signal_handler_uninstall_wrapper;
signal_handler_uninstall_wrapper.RegisterInitHandler(SignalHandlers::InstallSignalHandler,"SendTermSignal_Unset_Test", false);
signal_handler_uninstall_wrapper.RegisterShundownHandler(SignalHandlers::UninstallSignalHandler);
```
And the the method `RegisterShundownHandler` can be written with `Variable parameter template` like `Init` that user can pass the `shundown func` with any parameter.